lib/teams/server/message-actions/chat-message-actions.js

```diff
@@ -0,0 +1,235 @@
+const { sc, Logger } = require('@reldens/utils');
+const { ChatConst } = require('../../../chat/constants');
+const { TeamsConst } = require('../../constants');
+
+class ChatMessageActions
+{
+    constructor(props)
+    {
+        this.events = sc.get(props, 'events', false);
+        this.chatPlugin = sc.get(props, 'chatPlugin', false);
+        this.client = false;
+        if (!this.events) {
+            Logger.error('EventsManager undefined in Teams ChatMessageActions');
+        }
+        if (!this.chatPlugin) {
+            Logger.error('ChatPlugin undefined in Teams ChatMessageActions');
+        }
+    }
+
+    listenEvents()
+    {
+        this.inviteTeamAcceptedEventListener();
+        this.inviteTeamRejectedEventListener();
+        this.teamMemberLeaveEventListener();
+        this.inviteClanAcceptedEventListener();
+        this.inviteClanRejectedEventListener();
+        this.clanMemberLeavingEventListener();
+    }
+
+    inviteTeamAcceptedEventListener()
+    {
+        this.events.on('reldens.afterPlayerJoinedTeam', (props) => {
+            let playerJoiningSchema = sc.get(props, 'playerJoiningSchema', false);
+            if (!playerJoiningSchema) {
+                return false;
+            }
+            let currentTeam = sc.get(props, 'currentTeam', false);
+            if (!currentTeam) {
+                return false;
+            }
+            let playerJoiningName = playerJoiningSchema.playerName;
+            let message = this.createMessage(
+                TeamsConst.CHAT.MESSAGE.INVITE_ACCEPTED,
+                playerJoiningName
+            );
+            this.sendMessage(message, 'Team', currentTeam.ownerClient);
+            let messageEnter = this.createMessage(TeamsConst.CHAT.MESSAGE.ENTER,
+                playerJoiningName,
+                'team');
+            let otherPlayers = this.gatherOtherPlayersOnTeam(currentTeam, [playerJoiningSchema.player_id, currentTeam.owner.player_id]);
```

Comment:
remove the gatherOtherPlayersOnTeam, check TeamUpdatesHandler it can be done at once without the two extra loops